packages/react-devtools-shared/src/backend/fiber/renderer.js

```diff
@@ -1448,6 +1504,18 @@ export function attach(
         fiberToFiberInstanceMap.delete(alternate);
       }
     }
+
+    // TODO: This is not enough if this Fiber was filtered since we don't end up
+    // untracking it. We could use a WeakMap but that doesn't work for Paper tags.
```

Comment:
This is not really new. I discovered this was already an issue with leaking HostHoistables but it's really a blocker for landing this PR since it leaks DOM nodes if they're filtered which is way more common.

I can address this with a separate PR though which I think will need to start tracking some filtered Fiber Instances in some cases.

Reply:
Could this be fixed by just manually releasing host instance / resource if instance is filtered out?

Reply:
Yea I'm just not sure how to structure it yet because we can't just recursively do it because we also recursively walk the Instance tree. I also have another need to track FiberInstances that are filtered to get the HostInstance from a VirtualInstance. So I was going to see where that landed first.

I won't land this until it has a solution though.

Reply:
I went with this approach to manually release if they're filtered out:

https://github.com/facebook/react/pull/30831/commits/a4ae4d6eefbd311f5c6938003be28ebc4ad1c270

This will work with #30865 because we can start from the filtered fiber instance and then untrack from there.